test/test_time.cpp

```diff
@@ -50,6 +50,84 @@ class TestTimeFixture : public ::testing::Test
   }
 };
 
+// Tests the rcutils time unit conversion macros.
+TEST_F(TestTimeFixture, test_rcutils_time_conversion_macros) {
+  // Note: 9007199254740993 or higher cannot be represented anymore by intermediate type double
+  // without a loss of precision.
+
+  // seconds to nanoseconds
+  EXPECT_EQ(static_cast<int64_t>(RCUTILS_S_TO_NS(1)), 1000000000ll);  // int
+  EXPECT_EQ(static_cast<int64_t>(RCUTILS_S_TO_NS(0.2)), 200000000ll);  // double
```

Comment:
For this (and all of the other double checks), it seems to me that instead of casting the result of the macro to int64_t, we should actually change the expected value to be a double.  After all, that would more closely test that the output of the macro returns the type we expect, no?

Reply:
That is clearly beyond scope of this pull request, I think. Why would you expect the macros to return seconds as `double`? They don't. They return the same type as their argument, and eventually promote the integer to the width of a `long long` because of the `ll` suffix of the constants.

But `RCUTILS_MS_TO_S(1)` for example returns `0`, even if you would cast the result to `double` afterwards. Calling the macros with a `uint64_t` triggers a sign-conversion warning.


If you expect certain fixed input and output types, then they should probably be replaced by inline functions or the macros need some additional explicit casts. But that would be a potentially breaking change.

Reply:
I am sorry, I associated your comment with the wrong line of code. Clearly, if the argument is of type `double`, the output can be expected of type `double`, too, and the cast to `int64_t` can be omitted.

Reply:
> I am sorry, I associated your comment with the wrong line of code. Clearly, if the argument is of type `double`, the output can be expected of type `double`, too, and the cast to `int64_t` can be omitted.

Yeah, that's what I was getting at.  Sorry if I wasn't clear.  Anyway, could you please make that change?  Thanks.

Reply:
My thought was that the macro might be used to assign to an `int64_t`, like
```
double seconds = 42.5;
int64_t nanoseconds = RCUTILS_S_TO_NS(seconds);
```

But I could not find any usage of these macros with a floating point argument and this code snippet triggers a `-Wconversion` compiler warning, so probably they were just never meant for that purpose.